src/main/java/org/elasticsearch/action/admin/indices/status/TransportIndicesStatusAction.java

```diff
@@ -204,18 +197,14 @@ protected ShardStatus shardOperation(IndexShardStatusRequest request) throws Ela
                     default:
                         stage = PeerRecoveryStatus.Stage.INIT;
                 }
-                shardStatus.peerRecoveryStatus = new PeerRecoveryStatus(stage, peerRecoveryState.getTimer().startTime(),
-                        peerRecoveryState.getTimer().time(),
-                        peerRecoveryState.getIndex().totalByteCount(),
-                        peerRecoveryState.getIndex().reusedByteCount(),
-                        peerRecoveryState.getIndex().recoveredByteCount(), peerRecoveryState.getTranslog().currentTranslogOperations());
-            }
-
-            IndexShardGatewayService gatewayService = indexService.shardInjectorSafe(request.shardId().id()).getInstance(IndexShardGatewayService.class);
-            RecoveryState gatewayRecoveryState = gatewayService.recoveryState();
-            if (gatewayRecoveryState != null) {
+                shardStatus.peerRecoveryStatus = new PeerRecoveryStatus(stage, recoveryState.getTimer().startTime(),
+                        recoveryState.getTimer().time(),
+                        recoveryState.getIndex().totalBytes(),
```

Comment:
instead of calling `recoveryState.getIndex()` all time can we assign it to a local var above?